src/Google.Api.Gax/Api/Gax/BackoffSettings.cs

```diff
@@ -0,0 +1,116 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using System;
+
+namespace Google.Api.Gax
+{
+    /// <summary>
+    /// Backoff settings used within <see cref="RetrySettings"/> to implement
+    /// exponential backoff.
+    /// </summary>
+    public sealed class BackoffSettings
+    {
+        private TimeSpan _delay;
+        private TimeSpan _maxDelay;
+        private double _delayMultiplier = 1.0;
+
+        /// <summary>
+        /// The initial delay, either for the first retry or as the initial RPC timeout.
+        /// </summary>
+        /// <remarks>
+        /// The default value of this property is <see cref="TimeSpan.Zero"/>.
+        /// </remarks>
+        /// <exception cref="ArgumentOutOfRangeException">The value is negative.</exception>
+        public TimeSpan Delay
+        {
+            get { return _delay; }
+            set
+            {
+                if (value < TimeSpan.Zero)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value), value, "Delay must not be negative");
+                }
+                _delay = value;
+            }
+        }
+
+        /// <summary>
+        /// The multiplier to apply to the delay on each iteration; must be greater than or equal to 1.0.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// The default value of this property is 1.0.
+        /// </para>
+        /// <para>
+        /// As an example, a multiplier of 2.0 with a delay of 0.1s on an RPC which fails three times before
+        /// succeeding would lead to an initial delay between the first response and the second request
+        /// of 0.1s; a delay between the second response and the third request of 0.2s, and a delay
+        /// between the third response and the fourth request.
+        /// </para>
+        /// </remarks>
+        /// <exception cref="ArgumentOutOfRangeException">The value is not a real number, or is less than 1.</exception>
+        public double DelayMultiplier
+        {
+            get { return _delayMultiplier; }
+            set
+            {
+                if (value < 1 || double.IsInfinity(value) || double.IsNaN(value))
```

Comment:
nit: 1.0 as it's a double?
nit: Don't check for infinity here? Something like Double.MaxValue would have same effect as infinity, so why single out infinity to be invalid?